### ISSUE TITLE
CanSaveUnit should be a Test Any event, not a Test All event

### ISSUE THREAD

**Issue by Chrisy15** [LoneGazebo/Community-Patch-DLL#10815]
https://forums.civfanatics.com/threads/cansaveunit-is-flawed-introducing-cansaveunitgood.689112/

Tl;dr: CanSaveUnit saves a unit when function says `true`, but because it's an All event all subscribed functions have to return `true` for the unit to be saved. This is inconsistent with how other test events work, as they only require a single function to want the unnatural outcome for said outcome to be brought about. This issue can be solved by making CanSaveUnit a Test Any event, as then the single function which wants to change the expected outcome does not get overridden by any other functions which do not want to change the expected outcome. Treating these failed functions as though they were never there is important to avoid the functions subscribed to a test event from interfering with each other.

This issue needs to be solved on a DLL level, and I cannot think of a compatibility reason for keeping it as an All event or any advantage to it remaining as an All event.

It's possible that there are more Test All events which use `true` returns as their output-of-action which would similarly want to be made into Test Any events, but I haven't looked into any such thing. Would be worth checking over to fix them all in one go, I suppose.
